polytracker/src/CMakeLists.txt

```diff
@@ -50,7 +35,21 @@ set(TAINTDAG_SOURCES
     ${TAINTDAG_DIR}/fnmapping.cpp
     ${TAINTDAG_DIR}/fntrace.cpp)
 
-add_library(Polytracker STATIC ${POLY_SOURCES} ${TAINT_SOURCES}
+add_library(Polytracker STATIC ${POLYTRACKER_SOURCES} ${TAINT_SOURCES}
                                ${TAINTDAG_SOURCES})
+
+set(POLYTRACKER_COMPILE_OPTS -fno-exceptions -fno-rtti -Wall -Werror
+                             -stdlib=libc++ -fPIC)
+
+if(XRAY_ON)
+  list(APPEND POLYTRACKER_COMPILE_OPTS -fxray-instrument
+       -fxray-instruction-threshold=1)
+endif()
+
+target_include_directories(Polytracker PRIVATE ${POLYTRACKER_CXX_INCLUDE})
+target_compile_options(Polytracker PRIVATE ${POLYTRACKER_COMPILE_OPTS})
+target_link_options(Polytracker PRIVATE -stdlib=libc++)
```

Comment:
Should it be `PUBLIC`? We want to use the same stdlib across the entire final binary, right? Maybe that will happen anyway somehow?
If we made it public here (and the `target_compile_options` as well), I believe we wouldn't have to add e.g. https://github.com/trailofbits/polytracker/pull/6498/files#diff-ca270b0ff3ff8ec2df04556b61ade25034e5d38c1c0928f14db37bb63b8d4d8fR18-R21

Reply:
You're probably right, but part of the point of this PR is to make flag settings as local as possible, because our requirements for code is different in different places e.g. `passes` vs `taintdag`. IIRC there's a better way to do the build that @pgoodman suggested before, which I want to do once I get to moving DFSAN from `polytracker-llvm` to `polytracker`. This PR is meant to just fix the most pressing matters.